src/Hl7.Fhir.Core/Rest/SearchParams.cs

```diff
@@ -177,6 +183,20 @@ private static void addNonEmpty(string paramName, IList<string> values, string v
             values.Add(value);
         }
 
+        private static void addNonEmptyInclude(string paramName, IList<Tuple<string,List<string>>> values, string value)
+        {
+            if (String.IsNullOrEmpty(value)) throw Error.Format("Invalid {0} value: it cannot be empty".FormatWith(paramName));
+            if ( paramName == SEARCH_PARAM_INCLUDE)
+            {
+                values.Add(new Tuple<string, List<string>>(value, new List<string>()));
+            }
+            else // ITERATIVE INCLUDE
+            {
+                if ( !values.Any() ) throw Error.Format("Invalid {0}: must be preceded by a regular include".FormatWith(paramName));
+                values[values.Count - 1].Item2.Add(value);
```

Comment:
I don't think this is correct either: an `_include:iterate` do not have to correspond to the last `_include` in the parameters list, e.g.

```
GET [base]/MedicationDispense?_include=MedicationDispense:prescription  &_include=MedicationDispense:patient&_include:iterate=MedicationRequest:performer&criteria...
```

the `_include:iterate` correspond to the first `_include`, not the second